### ISSUE TITLE
VIP: flag to omit extcodesize checks

### ISSUE THREAD

**Issue by charles-cooper** [vyperlang/vyper#2550]
## Simple Summary
Add a flag to all external calls which results in the `extcodesize`/`returndatasize` check being dropped

## Motivation
"there are plenty of contracts that only interact with a set of whitelisted or validated at constructor time contracts which are wasting a good amount of gas on these extcodesize checks that will never fail"
\- @transmissions11

## Specification
Add a kwarg like `omit_contract_check` to all external calls. It defaults to `False` (keeping the current behavior) Ex.
```python
external_contract: ExternalContract

...

def my_func():
  self.external_contract.foo(..., omit_contract_check=True)
```

## References
Add any references that this VIP might reference (other VIPs/issues, links to blog posts, etc.)

## Copyright
Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/)

**Comment by fubuloubu** [comment]
Into this. adding to #2540 @transmissions11

**Comment by fubuloubu** [comment]
You know... on second thought, this is such a small feature and not really hurting anyone in any way, so I'm comfortable approving this one async

**Comment by charles-cooper** [comment]
I think we have enough consensus about this offline that we should go for it, maybe the only question is if the argument name - `omit_contract_check` - is any good

**Comment by charles-cooper** [comment]
Closed in #2551
